internal/controllers/synthetic_devices_controller.go

```diff
@@ -433,6 +431,45 @@ func (vc *SyntheticDevicesController) handleDeviceAPIIntegrationCreation(ctx con
 		mb, _ := json.Marshal(meta)
 		udi.Metadata = null.JSONFrom(mb)
 		udi.ExternalID = null.StringFrom(externalID)
+	case constants.TeslaVendor:
+		teslaID, err := strconv.Atoi(req.Credentials.ExternalID)
+		if err != nil {
+			vc.log.Err(err).Msgf("unable to parse external id %+v as integer", req.Credentials.ExternalID)
+			return err
+		}
+
+		if _, err := vc.teslaService.GetVehicle(req.Credentials.AccessToken, teslaID); err != nil {
+			vc.log.Err(err).Msg("unable to retrieve vehicle from Tesla")
+			return err
+		}
+
+		encAccess, err := vc.cipher.Encrypt(req.Credentials.AccessToken)
+		if err != nil {
+			return opaqueInternalError
+		}
+		encRefresh, err := vc.cipher.Encrypt(req.Credentials.RefreshToken)
+		if err != nil {
+			return opaqueInternalError
+		}
+		udi.AccessToken = null.StringFrom(encAccess)
+		udi.AccessExpiresAt = null.TimeFrom(time.Unix(req.Credentials.ExpiresIn, 0))
```

Comment:
In the OAuth2 spec, `expires_in` is the amount of time you have until the credentials expire, in seconds starting from "now". So we probably want to do some addition here.

Reply:
Why are we setting this twice? That's odd.